### PR TITLE
Removed all unused imports and other bindings.

### PR DESCRIPTION
**Pull request by Rembane**
If everything has gone as planned this should be a very boring commit.

### CODE REVIEW

**Review by adamse — APPROVED** [review]
God ide :)

**Comment by adamse** [comment]
```
== Section library ==
Redundant build-depends entry
* errors
```

from the build.

Another thing we can do is compile with `-Wall`.

**Comment by Rembane** [comment]
@adamse I added `-Wall` to the cabal-file. I hope this works. :D

**Comment by Jassob** [comment]
-Wall is almost always a good idea (some warnings are a bit frustrating, e.g. shadowing) but most are good 👍

File: mat-chalmers.cabal [hunk 1]
```diff
@@ -14,6 +14,7 @@ data-files:
 library
   default-language: Haskell2010
   hs-source-dirs: src
+  ghc-options: -Wall -fwarn-unused-binds -fwarn-unused-imports
```
Comment (adamse) [inline mat-chalmers.cabal:17]: I think modern ghc would rather you -Wunused-binds and -Wunused-imports, though both work.

**Comment by Rembane** [comment]
If it gets too obnoxious I'll remove it. :D

File: mat-chalmers.cabal (hunk 1 above)
Comment (adamse) [reply mat-chalmers.cabal:17]: So no need to change anything, just sharing knowledge :)

**Review by adamse — APPROVED** [review]

File: mat-chalmers.cabal (hunk 1 above)
Comment (Rembane) [reply mat-chalmers.cabal:17]: Thank you! I'm adding this to the next PR, it would be a shame to let knowledge go to waste. :)

Comment (adamse) [reply mat-chalmers.cabal:17]: https://downloads.haskell.org/~ghc/8.6.3/docs/html/users_guide/using-warnings.html#options-sanity

**Pull request merged**